### PR TITLE
Test Coverage 4532

### PR DESCRIPTION
**Pull request by hmcts-shajahan**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PRL-4532

### Change description ###

Adding testing covergae to the following classes:

'**/uk/gov/hmcts/reform/prl/controllers/citizen/FeesAndPaymentCitizenController.java',
'**/uk/gov/hmcts/reform/prl/controllers/testingsupport/TestingSupportController.java',
'**/uk/gov/hmcts/reform/prl/services/FeeService.java',
'**/uk/gov/hmcts/reform/prl/controllers/UploadAdditionalApplicationController.java',
'**/uk/gov/hmcts/reform/prl/services/ApplicationsFeeCalculator.java'

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/prl/services/FeeService.java [hunk 1]
```diff
@@ -211,18 +211,18 @@ private static Optional<FeeType> fromOtherPartyConsentAndHearing(String otherPar
...
-            } else if (notice.equals(YES)) {
-                return C2_WITH_NOTICE;
-            } else {
-                return null;
+            if (isNotBlank(notice)) {
+                if (notice.equals(NO)) {
+                    return C2_WITHOUT_NOTICE;
+                } else if (notice.equals(YES)) {
+                    return C2_WITH_NOTICE;
+                }
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/services/FeeService.java:222]: Always compare with the Constant, then you can avoid null pointer exception as well as an extra if block.
So it should be - if (YES.equals(otherPartyConsent)) and so on.

File: src/test/java/uk/gov/hmcts/reform/prl/controllers/TestingSupportControllerTest.java [hunk 2]
```diff
@@ -153,6 +153,28 @@ public void testExeptionForSubmittedRespondentTaskList() {
...
+        verify(testingSupportService, times(1)).initiateCaseCreationForCourtNav(Mockito.anyString(), Mockito.any(CallbackRequest.class));
+    }
+
+    @Test
+    public void testExeptionForaboutToSubmitCaseCreationCourtNav() {
+        Mockito.when(authorisationService.isAuthorized(authToken, s2sToken)).thenReturn(false);
+        assertExpectedException(() -> {
+            testingSupportController.aboutToSubmitCaseCreationCourtNav(authToken, s2sToken, callbackRequest);
+        }, RuntimeException.class, "Invalid Client");
+    }
```
Comment (AlokDatta) [inline src/test/java/uk/gov/hmcts/reform/prl/controllers/TestingSupportControllerTest.java:169]: move this file within a package testingSupport similar to main folder

**Review by AlokDatta — CHANGES_REQUESTED** [review]
Minor comments, please address.

File: src/main/java/uk/gov/hmcts/reform/prl/services/FeeService.java (hunk 1 above)
Comment (hmcts-shajahan) [reply src/main/java/uk/gov/hmcts/reform/prl/services/FeeService.java:222]: Done

File: src/test/java/uk/gov/hmcts/reform/prl/controllers/TestingSupportControllerTest.java (hunk 2 above)
Comment (hmcts-shajahan) [reply src/test/java/uk/gov/hmcts/reform/prl/controllers/TestingSupportControllerTest.java:169]: Resolved

File: src/main/java/uk/gov/hmcts/reform/prl/services/FeeService.java [hunk 3]
```diff
@@ -211,18 +211,17 @@ private static Optional<FeeType> fromOtherPartyConsentAndHearing(String otherPar
...
 
-        if (otherPartyConsent.equals(YES)) {
+        if (YES.equals(otherPartyConsent)) {
             return C2_WITHOUT_NOTICE;
         } else {
-            if (notice.equals(NO)) {
+            if (NO.equals(notice)) {
                 return C2_WITHOUT_NOTICE;
-            } else if (notice.equals(YES)) {
+            } else  if (YES.equals(notice)) {
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/services/FeeService.java:219]: extra space.

**Review by AlokDatta — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/prl/services/FeeService.java (hunk 3 above)
Comment (hmcts-shajahan) [reply src/main/java/uk/gov/hmcts/reform/prl/services/FeeService.java]: Resolved

**Review by AlokDatta — APPROVED** [review]

**Review by shashisk97 — APPROVED** [review]
LGTM

**Pull request merged**